feat: Add GTM event tracking to download buttons

PanchoBubble opened the pull request:
This commit adds Google Tag Manager (GTM) event tracking to the download buttons for MacOS, Windows, and Linux. When a user clicks on a download button, a `download_button_clicked` event is sent to GTM, along with the platform (macos, windows, or linux) that was clicked. This will allow us to track which platforms are being downloaded most frequently.

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

- **New Features**
  - Added event tracking for download button clicks on different operating systems. Users' interactions with Mac, Windows, and Linux download buttons are now tracked for analytics purposes.

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

Krakaw commented on _includes/downloads.html:
```diff
@@ -70,18 +70,20 @@ <h2 class="dl-title">{{ os.title }}</h2>
                     <div class="checksum">
-                        SHA256: {{ os.checksum }} 
-                        
+                        SHA256: {{ os.checksum }}
```
This changes with every file, it needs to come from the API

Krakaw commented on _includes/downloads.html:
```diff
@@ -35,12 +35,12 @@ <h2 class="dl-title">{{ os.title }}</h2>
-                    {% if os.download %}
-                    <a class="btn " href="{{ site.baseurl}}/{{ os.download }}">{{os.ctaText}}</a>
+                    {% if os.downloadBtnID %}
+                    <a class="btn" id="{{ os.downloadBtnID }}">{{os.ctaText}}</a>
```
Where's the href?

shanimal08 replied:
It gets added in in the JS on line 81 😬

PanchoBubble committed:
feat: Add GTM event tracking to download buttons

This commit adds Google Tag Manager (GTM) event tracking to the download buttons for MacOS, Windows, and Linux. When a user clicks on a download button, a `download_button_clicked` event is sent to GTM, along with the platform (macos, windows, or linux) that was clicked. This will allow us to track which platforms are being downloaded most frequently.
